### PR TITLE
Added options to set the listen address for standalone mode

### PR DESCRIPTION
**Pull request by jeffallen**
This might help with the problem mentioned in certbot/certbot#255.

### CODE REVIEW

File: certbot/cli.py [hunk 1]
```diff
@@ -991,10 +991,18 @@ def prepare_and_parse_args(plugins, args, detect_defaults=False):  # pylint: dis
         default=flag_default("tls_sni_01_port"),
         help=config_help("tls_sni_01_port"))
     helpful.add(
+        ["testing", "standalone", "apache", "nginx"], "--tls-sni-01-addr",
```
Comment (zjs) [inline certbot/cli.py]: This line seems to indicate that the `--tls-sni-01-addr` argument is applicable to the standalone, apache, and nginx plugins. However, this CL only makes use of it in the standalone plugin (and that seems to be intentional).

File: certbot/cli.py [hunk 2]
```diff
@@ -991,10 +991,18 @@ def prepare_and_parse_args(plugins, args, detect_defaults=False):  # pylint: dis
...
+        default=flag_default("tls_sni_01_addr"),
+        help=config_help("tls_sni_01_addr"))
+    helpful.add(
         ["testing", "standalone", "manual"], "--http-01-port", type=int,
         dest="http01_port",
         default=flag_default("http01_port"), help=config_help("http01_port"))
     helpful.add(
+        ["testing", "standalone", "manual"], "--http-01-addr",
```
Comment (zjs) [inline certbot/cli.py]: This line seems to indicate that the `--http-01-addr` argument is applicable to the standalone and manual plugins.  However, this CL only makes use of it in the standalone plugin (and that seems to be intentional).

Comment (jeffallen) [inline certbot/cli.py]: Honestly, I don't know either. I copied and pasted the option declaration for --http-01-port without understanding that part of it. @pde touched that line last, perhaps he can tell us how this should look?

File: docs/contributing.rst [hunk 3]
```diff
@@ -99,6 +99,13 @@ specific failures when they happen. Also note that the full test suite
 will attempt to modify your system's Apache config if your user has sudo
 permissions, so it should not be run on a production Apache server.
 
+Running the tests on Ubuntu requires some extra packages to be installed.
+You can install them like this:
+
+.. code-block:: shell
+
+   sudo apt install libaugeas-dev python2.7-dev python3-dev
```
Comment (bmw) [inline docs/contributing.rst]: Other than `python3-dev`, these dependencies are installed by following the instructions in the [Getting Started](https://certbot.eff.org/docs/contributing.html#running-a-local-copy-of-the-client) section of our developer guide. I created #4735 to track that issue.

If you wanted, you could add something to the top of this section saying something like:
```
If you're not running the client using our :ref:`Docker setup <docker>`, make sure you follow the instructions at the beginning of the :ref:`Getting Started <getting_started>` section to ensure Certbot and it's dependencies are properly installed.
```
If you did something like this, I think it should be in a separate PR though.

**Comment by jeffallen** [comment]
Closing/opening to trigger new travis build.

File: tests/travis-integration.sh [hunk 4]
```diff
@@ -6,7 +6,7 @@ set -o errexit
 
 source .tox/$TOXENV/bin/activate
 
-until curl http://boulder:4000/directory 2>/dev/null; do
+until curl http://localhost:4000/directory 2>/dev/null; do
```
Comment (bmw) [inline tests/travis-integration.sh]: I'd like to see this change removed. Per the name of this script, it was written to run in Travis where the informative name `boulder` works properly. To run integration tests locally, you should follow the instructions found at [Integration testing with the Boulder CA](https://certbot.eff.org/docs/contributing.html#integration-testing-with-the-boulder-ca).

Comment (jeffallen) [inline tests/travis-integration.sh]: Sorry, that was a mistaken checkin.

**Comment by bmw** [comment]
LGTM. Thanks @jeffallen.

**Pull request merged**